nibabel/nicom/utils.py

```diff
@@ -27,26 +27,20 @@ def find_private_section(dcm_data, group_no, creator):
     Returns
     -------
     element_start : int
-        Element number at which named section starts
+        Element number at which named section starts.
     """
-    is_regex = hasattr(creator, 'search')
-    if not is_regex:  # assume string / bytes
+    if hasattr(creator, 'search'):
+        match_func = lambda x : creator.search(x)
+    else:  # assume string / bytes
         creator = asstr(creator)
```

Comment:
On the whole, I think we should be aiming to remove `py3k` functions. Is it safe to say that `creator` has the same type as `element.value`, or can there be mixed-and-matched `str`/`bytes`? Or is `element.value` at least always bytes or always str?

Reply:
To be clear, this does not need to be addressed in this PR, but when we actually know the types, we should work with them directly. `asstr()` says to me "here be dragons".

Reply:
I don't think `creator` need have the same type as `element.value` because we are testing all `element` values, and only some of them may be specifying offsets (though maybe not-offset elements would be an error).

Reply:
I agree the `asstr` is worrying, and in this case, it's because I do not know why `.value` will sometimes be `bytes` and sometimes `str`.   I should investigate, but it would take a little while to determine why differences would arise, and be sure that these won't arise in the wild.

Reply:
Okay. Let's leave it alone for now.